flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -182,4 +219,102 @@ public FileStoreSourceSplitSerializer getSplitSerializer() {
     public PendingSplitsCheckpointSerializer getEnumeratorCheckpointSerializer() {
         return new PendingSplitsCheckpointSerializer(getSplitSerializer());
     }
+
+    private void writeObject(ObjectOutputStream out) throws IOException {
```

Comment:
Could we just introduce two separate class named `PartMeta` and `BucketMeta` to manage all the `DataFileMetaData` for a given partition and a given bucket, and moving this serializer & deserializer logic into those two classes ?